easybuild/easyconfigs/c/CUDAcompat/CUDAcompat-11.7.eb

```diff
@@ -0,0 +1,33 @@
+name = 'CUDAcompat'
+version = '11.7'
+nv_version = '515.48.07'
+hidden = True
+
+# Compatible driver versions:
+# 450.36.06+ (CUDA 11.0)
+# 470.57.02+ (CUDA 11.4)
+# 510.39.01+ (CUDA 11.6)
```

Comment:
Can we do something with this information? Perhaps set it as a list and have the easyblock check the "true" driver version against the compatible versions so we can verify if the compat libraries will actually work?

Reply:
I was thinking about this but decided against it: Getting the "true" driver version isn't trivial to do portably and you may install on another machine than the one you'll run on.

Reply:
You can get the driver version with
```
nvidia-smi --query-gpu=driver_version --format=csv,noheader
```
which reports the same value regardless of the compat libraries used (in my testing at least).

But of course if you don't have a GPU...

Reply:
I'll add a check to the EasyBlock. I think the test-step is a good place as that allows skipping it.
Similar to the TensorFlow EasyBlock I'll skip the check if `nvidia-smi` isn't there.

Reply:
I'm not sure on that front, it's something I've only been able to see through testing with a CUDA executable. In my case I have access to a system with 460.73.01 drivers, all the 11.7 compat libraries fail to run CUDA executables (even though `nvidia-smi` reports the CUDA version just fine) with the documented error code:
```
cudaGetDeviceCount returned 803
```
The driver I have doesn't appear in the list at https://docs.nvidia.com/deploy/cuda-compatibility/index.html#use-the-right-compat-package at all.

Reply:
Well that implies that the above approach would work fine: Your 460.x driver would be reported as incompatible with the compat libraries and your tests seem to confirm that it really is incompatible.

Reply:
Not quite, the `460.73.01` driver is also not listed for the `11.6` compat libraries, but if I test it with those it works just fine. That just says to me the only way you can really know is by actually compiling something with `nvcc` and trying to run it.

Reply:
I tested the 11.6 compat libs on 460.32.03 and that failed with 804 (cudaErrorCompatNotSupportedOnDevice) similar to your 803 (cudaErrorSystemDriverMismatch)

> That just says to me the only way you can really know is by actually compiling something with nvcc and trying to run it.

I'm not sure how to do that. I want the compat module as a dependency of the (some) CUDA module so when users load CUDA they get the right compat package. But if you need to load CUDA to test the compat package you have a cyclic dependency. Hence the idea to do that test in the CUDA easyblock. Because that is what you'd like to have working.